src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] The `die` method contains unnecessary conditional checking of the animal's health. Since this method is only called when an animal's health is already less than or equal to 0, the condition `if (this.health <= 0)` is redundant. You can directly use the filter method without this check.